python/lsst/cp/pipe/defects.py

```diff
@@ -367,27 +393,91 @@ def _writeData(self, dataRef, defects):
         defects : `lsst.meas.algorithms.Defect`
             The defects to be written.
         """
-        filename = dataRef.getUri(write=True)  # does not guarantee that full path exists
-        dirname = os.path.dirname(filename)
-        if not os.path.exists(dirname):
-            os.makedirs(dirname)
+        date = str(dafBase.DateTime(midTime, dafBase.DateTime.MJD).toPython())
+        date = date.replace(" ", "T")  # original has a space in the middle
+
+        detName = self._getDetectorName(dataRef)
+        instrumentName = self._getInstrumentName(dataRef)
+        detNum = self._getDetectorNumber(dataRef)
+        if not self.config.ignoreFilters:
+            filt = self._filterSetToFilterString(filters)
+        else:
+            filt = self.config.nullFilterName
+
+        CALIB_ID = f"detectorName={detName} detector={detNum} calibDate={date} ccd={detNum} filter={filt}"
+        try:
+            raftName = detName.split("_")[0]
```

Comment:
Do all instruments include the raft name in detName but also need a separate raftName?

Reply:
No (hence the `try`), plus we're actually using `detName` totally wrong at the moment, but I can only try to fix so many things at once 😔

Reply:
Something I thought of but forgot to ask: does this need to modify detName if a raftName is found, or is something like `detName='R00_S00', raftName='R00'` expected?

Reply:
It's not expected, as such, but that is indeed exactly the "using `detName` totally wrong" that I refer to above. `detName` should only _ever_ be `S12`-like, but that's not how it is right now 😔